great_expectations/datasource/data_connector/configured_asset_azure_data_connector.py

```diff
@@ -1,17 +1,160 @@
+        azure_options: Optional[dict] = None,
+        batch_spec_passthrough: Optional[dict] = None,
+    ):
+        """
+        ConfiguredAssetDataConnector for connecting to Azure.
+
+        Args:
+            name (str): required name for DataConnector
+            datasource_name (str): required name for datasource
+            container (str): container name for Azure Blob Storage
+            assets (dict): dict of asset configuration (required for ConfiguredAssetDataConnector)
+            execution_engine (ExecutionEngine): optional reference to ExecutionEngine
+            default_regex (dict): optional regex configuration for filtering data_references
+            sorters (list): optional list of sorters for sorting data_references
+            name_starts_with (str): Azure prefix
+            delimiter (str): Azure delimiter
+            azure_options (dict): wrapper object for **kwargs
+            batch_spec_passthrough (dict): dictionary with keys that will be added directly to batch_spec
+        """
+        logger.debug(f'Constructing ConfiguredAssetAzureDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            assets=assets,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+        self._container = container
+        self._name_starts_with = os.path.join(name_starts_with, "")
+        self._delimiter = delimiter
+
+        if azure_options is None:
+            azure_options = {}
+
+        # Thanks to schema validation, we are guaranteed to have one of `conn_str` or `account_url` to
+        # use in authentication (but not both). If the format or content of the provided keys is invalid,
+        # the assignment of `self._account_name` and `self._azure` will fail and an error will be raised.
+        conn_str = azure_options.get("conn_str")
+        account_url = azure_options.get("account_url")
+        assert bool(conn_str) ^ bool(account_url)
+
+        try:
+            if conn_str is not None:
+                self._account_name = re.search(
+                    r".*?AccountName=(.+?);.*?", conn_str
+                ).group(1)
+                self._azure = BlobServiceClient.from_connection_string(**azure_options)
+            elif account_url is not None:
+                self._account_name = re.search(
+                    r"(?:https?://)?(.+?).blob.core.windows.net", account_url
+                ).group(1)
+                self._azure = BlobServiceClient(**azure_options)
+        except (TypeError, AttributeError):
+            raise ImportError(
+                "Unable to load Azure BlobServiceClient (it is required for ConfiguredAssetAzureDataConnector). \
+                Please ensure that you have provided the appropriate keys to `azure_options` for authentication."
+            )
```

Comment:
![image](https://user-images.githubusercontent.com/49923762/128646668-135d6172-c28f-4c80-8a64-05af2a4a4a86.png)
While RegEx can be a dangerous tool, I think these `re.search()` calls are simple and effective enough to be used here. We need to be able to parse `account_name` from the various auth methods for usage in `_get_full_file_path` and this method allows us to do so without explicitly asking the user for yet another key in `azure_options`.